test/e2e/serial/config/config.go

```diff
@@ -67,3 +76,62 @@ func GetRteCiImage() string {
 	}
 	return nropTestCIImage
 }
+
+func CheckNodesTopology() error {
+	kconfigs, err := validator.GetKubeletConfigurationsFromTASEnabledNodes(context.TODO(), e2eclient.Client)
+	if err != nil {
+		return fmt.Errorf("error while trying to get KubeletConfigurations. error: %w", err)
+	}
+
+	var nrtList nrtv1alpha2.NodeResourceTopologyList
+
+	err = e2eclient.Client.List(context.TODO(), &nrtList)
+	if err != nil {
+		return fmt.Errorf("error while trying to get NodeResourceTopology objects. error: %w", err)
+	}
+
+	policies := []nrtv1alpha2.TopologyManagerPolicy{
+		nrtv1alpha2.SingleNUMANodeContainerLevel,
+		nrtv1alpha2.SingleNUMANodePodLevel,
+	}
+	nrts := e2enrt.FilterByPolicies(nrtList.Items, policies)
+	singleNumaNodeNumber := len(nrts)
+	if singleNumaNodeNumber < minNumberOfNodesWithRightTopology {
+		return fmt.Errorf("not enough nodes with valid policy - found %d, need at least %d", len(nrts), minNumberOfNodesWithRightTopology)
+	}
+
+	nrtsSingleNumaNodePodLevel := e2enrt.FilterTopologyManagerPolicy(nrts, nrtv1alpha2.SingleNUMANodePodLevel)
+	podLevelNumber := len(nrtsSingleNumaNodePodLevel)
+	containerLevelNumber := singleNumaNodeNumber - podLevelNumber
```

Comment:
I'd not just add this assumption here. Let's count explicitely, perhaps also raising a warning (= log)/error if the sum of containerLevel and podLevel is different than singleNumaNodeNumber